src/Microsoft.AspNet.Mvc.ViewFeatures/ViewExecutor.cs

```diff
@@ -36,7 +36,20 @@ public static class ViewExecutor
         {
             var response = actionContext.HttpContext.Response;
 
-            contentType = contentType ?? DefaultContentType;
+            if (contentType == null)
+            {
+                contentType = DefaultContentType;
+
+                if (!string.IsNullOrEmpty(response.ContentType))
+                {
+                    MediaTypeHeaderValue parsedContentType;
+                    if (MediaTypeHeaderValue.TryParse(response.ContentType, out parsedContentType))
+                    {
+                        contentType = parsedContentType;
+                    }
```

Comment:
Hmm. Just curious, we already check if the encoding is set in a user supplied content type. Why not check for invalid ones?